device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -575,7 +580,7 @@ public void sendEventAsync(Message message, IotHubEventCallback callback, Object
      * @throws IllegalArgumentException if the message provided is {@code null}.
      * @throws IllegalStateException if the client has not been opened yet or is already closed.
      */
-    public void sendEventAsync(List<Message> messages, IotHubEventCallback callback, Object callbackContext)
```

Comment:
Making this plural since it sends a batch of messages

Reply:
Does it really send them in a batch?

Reply:
For HTTP, yes. Just talked with Azad about this, and I'm considering making this API for HTTP only for now. Our current implementation is such that MQTT/AMQP clients just send each message individually, and this API isn't set up to handle partial successes/failures

Reply:
I'm very much against the "false" batch send we have in .NET for MQTT (and anywhere else). It is deceiving to the customer who thinks it is batch but really no better than a foreach they can implement.

And what happens if it fails at 61/100 messages? When an exception is thrown how will someone know which message to start resending and not get dupes on the service? I'd rather we just not include it than have partial/poor support for it.